action_server/src/sema4ai/action_server/_actions_run.py

```diff
@@ -222,34 +230,49 @@ def _run_action_in_thread(
                     reuse_process,
                 )
 
-                if returncode == 0:
+                try:
+                    run_result_str: str = result_json.read_text("utf-8", "replace")
+                except Exception:
+                    raise RuntimeError(
+                        "It was not possible to collect the contents of the "
+                        "result (json not created)."
+                    )
+
+                try:
+                    result_contents = json.loads(run_result_str)
+                except Exception:
+                    raise RuntimeError(
+                        f"Error loading the contents of {run_result_str} as json."
```

Comment:
```suggestion
                        f"Error loading the contents of {run_result_str!r} as json."
```
If you want it repr-ed.

Reply:
Given that it's an indented json I think it's better not to do that (so messages may be more well formatted to show in the terminal).